### DISCUSSION TITLE
fcm-make configuration for an LFRic-Apps suite on Monsoon

### DISCUSSION THREAD

**Discussion by dennissergeev** [MetOffice/simulation-systems#202] [Monsoon]
Hello,

I was wondering if anyone could help me with setting up fcm-make keywords for a standalone LFRic-Apps suite on Monsoon please. The suite is [`u-dg617`](https://code.metoffice.gov.uk/trac/roses-u/browser/d/g/6/1/7/trunk).

At the start of `fcm_make_lfric` task, I get the following error:
```
[FAIL] https://code.metoffice.gov.uk/svn/lfric_apps/main/trunk@vn1.0: revision keyword not defined

[FAIL] fcm make -f /scratch/d03/deser/cylc-run/u-dg617/run1/work/19880901T0000Z/fcm_make_lfric/fcm-make.cfg -C /home/d03/deser/cylc-run/u-dg617/run1/share/fcm_make_lfric -j 1 --ignore-lock # return-code=2
2024-05-30T11:42:10Z CRITICAL - failed/ERR
```

<details>
<summary>Contents of app/fcm_make_lfric/rose-app.conf </summary>

```
meta=fcm_make
mode=fcm_make
opt.jobs=1

[env]
lfric_apps_rev=@vn1.0
lfric_apps_sources=branches/pkg/Config/vn1.0_gc5lfric_package@1265
lfric_core_rev=@apps1.0
lfric_core_sources=branches/pkg/Config/apps1.0_gc5lfric_package@50299
```
</details>

I also tried setting `lfric_apps_rev=@782` but it results in

```
[FAIL] https://code.metoffice.gov.uk/svn/lfric_apps/main/trunk@782: not found
```

<details>
<summary>Contents of app/fcm_make_lfric/file/fcm-make.cfg </summary>

```
steps = extract

extract.ns = lfric_apps lfric_core

extract.location{primary}[lfric_apps] = fcm:lfric_apps.x
extract.location[lfric_apps] = trunk$lfric_apps_rev
extract.location{diff}[lfric_apps] = $lfric_apps_sources

extract.location{primary}[lfric_core] = fcm:lfric.x
extract.location[lfric_core] = trunk$lfric_core_rev
extract.location{diff}[lfric_core] = $lfric_core_sources
```
</details>

As a test, these commands work fine though, so I don't know what is wrong with the config files above:

```
fcm ls https://code.metoffice.gov.uk/svn/lfric_apps/main/trunk@vn1.0
fcm ls fcm:lfric_apps.x/trunk@vn1.0
```

Thanks,
Denis

**Comment by dpmatthews** [comment]
You can't access SRS (code.metoffice.gov.uk) repositories directly in tasks due to the need to authenticate. Instead you need to use mirrors which we maintain on Monsoon (updated every 5 minutes). To do this change `fcm:lfric.x` to `fcm:lfric.xm` and `fcm:lfric_apps.x` to `fcm:lfric_apps.xm`.
(I've just setup the lfric_apps mirror on Monsoon)

**Reply by dennissergeev** [reply]
Many thanks @dpmatthews! This solved the problem.

**Comment by dennissergeev** [comment]
@dpmatthews Another question if I may. Why do I get this error: `RAMTMP: unbound variable`?

It's used in the `mesh` and `lfric_atm` `compile` tasks in `flow.cylc`:
```
./flow.cylc:  {% set FASTMEM = "\$RAMTMP" %}
./flow.cylc:  {% set FASTMEM = "\$RAMTMP" %}
./flow.cylc:            WORKING_DIR          = $RAMTMP/mesh_tools
./flow.cylc:            WORKING_DIR          = $RAMTMP/lfric_atm
```

**Reply by dpmatthews** [reply]
`RAMTMP` is defined for PBS jobs which have requested a RAM drive: https://code.metoffice.gov.uk/doc/monsoon2/bestPractice.html#memory-resident-temporary-file-system-ram-drive
However, you are currently running these jobs as background jobs on the login node rather than via PBS.

**Reply by dennissergeev** [reply]
ah okay thanks!
